sample_tracking/app/login/signup.tsx

```diff
@@ -95,7 +105,41 @@ export default function SignUp(props: SignUpProps) {
         const name = `${formData.firstName} ${formData.lastName}`;
         const labValue = orgName ? availableOrgs[orgName] : '';
 
-        await createUserWithEmailAndPassword(auth, formData.email, formData.password);
+        await createUserWithEmailAndPassword(auth, formData.email, formData.password).catch((error) => {
+            console.log(error);
+            switch (error.code) {
+                case 'auth/weak-password':
+                    setErrorText({
+                        ...errorText,
+                        password: error.message.substring(error.message.indexOf(':') + 2, error.message.indexOf('('))
+                    })
+                    return;
+                case 'auth/email-already-exists':
+                    setErrorText({
+                        ...errorText,
+                        email: error.message.substring(error.message.indexOf(':') + 2, error.message.indexOf('('))
+                    })
+                    return;
+                case 'auth/email-already-in-use':
+                    setErrorText({
+                        ...errorText,
+                        email: t('emailAlreadyInUse')
+                    })
+                    return;
+                default:
+                    const cancelFunction = () => {
+                        setConfirmationBoxData(null);
+                    }
+                    const title = `${t('errorCreatingAccount')} + ${error.message}`;
```

Comment:
there probably is no way to get a localized version of error.message is there?

Reply:
According to this there isn't a way to get localized error messages but this should only be shown if its an unusual error so it won't be very common. 
https://stackoverflow.com/questions/42439504/firebase-error-messages-in-different-languages